internal/datastore/crdb/pool/pool.go

```diff
@@ -172,7 +172,11 @@ func (p *RetryPool) QueryFunc(ctx context.Context, rowsFunc func(ctx context.Con
 			return err
 		}
 		defer rows.Close()
-		return rowsFunc(ctx, rows)
+		err = rowsFunc(ctx, rows)
```

Comment:
there is at least one `rowsFunc` that I saw is not handling `rows.Err`. i thought it was sensible to do this check outside so as to safeguard against that _and_ future changes, but i'm open to changing each individual callee.

Reply:
I think we need to check for the error here first; if we get back an `err`, return that. Otherwise, check the `rows.Err()`